apps/server/src/providers/index.test.ts

```diff
@@ -0,0 +1,114 @@
+import { http, HttpResponse } from 'msw'
+import { server } from '../test/testServer'
+import { ProviderApiBase } from './index'
+import { createClient } from '../__mocks__/redis'
+import { ConnectionStatus, OAuthStatus } from '../../src/shared/contract'
+import { READ_MEMBER_STATUS_PATH } from '../test/handlers'
+import { MxApi } from './mx'
+
+const redisMock = createClient()
+const testConnectionId = 'test_connection_id'
+
+const mxApi = new MxApi(
+  {
+    mxProd: {
+      username: 'testUsername',
+      password: 'testPassword'
+    },
+    storageClient: redisMock
+  },
+  false
+)
+
+const providerApiBase = new ProviderApiBase({
+  context: {
+    provider: 'mx',
+    auth: {
+      token: 'test_token',
+      iv: 'test_iv'
+    },
+    resolved_user_id: 'test_user_id'
+  }
+})
+
+describe('ProviderApiBase', () => {
+  describe('getOauthState', () => {
+    beforeAll(async () => {
+      await providerApiBase.init()
+      providerApiBase.serviceClient = mxApi
+    })
+
+    describe('oauth status PENDING', () => {
+      it('returns the expected state', async () => {
+        server.use(
+          http.get(READ_MEMBER_STATUS_PATH, () =>
+            HttpResponse.json({
+              member: {
+                connection_status: 'PENDING',
+                guid: 'memberStatusGuid'
+              }
+            })
+          )
+        )
+
+        expect(await providerApiBase.getOauthState(testConnectionId)).toEqual({
+          oauth_state: {
+            guid: 'test_connection_id',
+            inbound_member_guid: 'test_connection_id',
+            outbound_member_guid: 'test_connection_id',
+            auth_status: OAuthStatus.PENDING
+          }
+        })
+      })
+    })
+
+    describe('oauth status COMPLETE', () => {
```

Comment:
Can you create a ticket that we can do later to write the rest of the tests for this file? I just don't want it to get forgotten because we finished some of the tests in here

Reply:
Good idea: https://universalconnect.atlassian.net/browse/UCP-133